frontend/handlers.go

```diff
@@ -103,8 +121,22 @@ var registeredHandlers = &handlerList{
 // This can be changed by calling [SetDefault].
 //
 // Registered handlers may be overridden by [dalec.Spec.Targets].
-func RegisterHandler(group string, t bktargets.Target, build BuildFunc) {
-	registeredHandlers.Add(group, &targetWrapper{Target: t, Build: build})
+func RegisterHandler(group string, t bktargets.Target, build BuildFunc, opts ...handlerOpt) {
+	var cfg handlerOpts
+	for _, f := range opts {
+		f(&cfg)
+	}
+
+	validations := []validateFunc{}
```

Comment:
We can't do this here.
Target registration needs to be completely separate from everything else.

I think its fine to have targets do extra validation on the spec as needed.

Reply:
> We can't do this here. Target registration needs to be completely separate from everything else.

Ok I'll change it.

Can you explain why? Helps me learn.

Reply:
Its mixing concerns.
Part of the routing logic changes removes this spec loading part as well, so the build targets need to load the spec themselves.

Reply:
done